src/main/webapp/app/shared/sidebar/sidebar.helper.ts

```diff
@@ -0,0 +1,269 @@
+import { DifficultyFilterOption, ExerciseCategoryFilterOption, ExerciseFilterOptions, ExerciseTypeFilterOption, FilterOption, RangeFilter } from 'app/types/exercise-filter';
+import { SidebarCardElement, SidebarData } from 'app/types/sidebar';
+import { ExerciseCategory } from 'app/entities/exercise-category.model';
+import { DifficultyLevel, ExerciseType, getIcon } from 'app/entities/exercise.model';
+import { getLatestResultOfStudentParticipation } from 'app/exercises/shared/participation/participation.utils';
+import { roundToNextMultiple } from 'app/shared/util/utils';
+
+const POINTS_STEP = 1;
+const SCORE_THRESHOLD_TO_INCREASE_STEP = 20;
+const SMALL_SCORE_STEP = 1;
+const SCORE_STEP = 5;
+
+const DEFAULT_DIFFICULTIES_FILTER: DifficultyFilterOption[] = [
+    { name: 'artemisApp.exercise.easy', value: DifficultyLevel.EASY, checked: false },
+    { name: 'artemisApp.exercise.medium', value: DifficultyLevel.MEDIUM, checked: false },
+    { name: 'artemisApp.exercise.hard', value: DifficultyLevel.HARD, checked: false },
+];
+
+const DEFAULT_EXERCISE_TYPES_FILTER: ExerciseTypeFilterOption[] = [
+    { name: 'artemisApp.courseStatistics.programming', value: ExerciseType.PROGRAMMING, checked: false, icon: getIcon(ExerciseType.PROGRAMMING) },
+    { name: 'artemisApp.courseStatistics.quiz', value: ExerciseType.QUIZ, checked: false, icon: getIcon(ExerciseType.QUIZ) },
+    { name: 'artemisApp.courseStatistics.modeling', value: ExerciseType.MODELING, checked: false, icon: getIcon(ExerciseType.MODELING) },
+    { name: 'artemisApp.courseStatistics.text', value: ExerciseType.TEXT, checked: false, icon: getIcon(ExerciseType.TEXT) },
+    { name: 'artemisApp.courseStatistics.file-upload', value: ExerciseType.FILE_UPLOAD, checked: false, icon: getIcon(ExerciseType.FILE_UPLOAD) },
+];
```

Comment:
Quite some refactoring to improve the readability. This is where the main changes happened since the last review.